nix_update/update.py

```diff
@@ -270,6 +270,104 @@ def update_cargo_lock(
                 print(line, end="")
 
 
+def update_pubspec_lock(opts: Options, filename: str) -> None:
+    with tempfile.TemporaryDirectory() as tempdir:
+        with Path(filename).open() as f:
+            match = re.search(
+                r"pubspecLock\s*=\s*lib\.importJSON\s+([^;]+)\s*;",
+                f.read(),
+            )
+            if match is None:
+                return
+            dst = Path(match.group(1))
+            if not dst.is_absolute():
+                dst = Path(filename).resolve().parent / dst
+                dst = dst.resolve()
+        if opts.generate_lockfile:
+            generate_lockfile(opts, filename, "pubspec")
+            with Path.open(
+                Path(tempdir) / "pubspec.lock.json",
+                "w",
+                encoding="utf-8",
+            ) as f:
+                run(
+                    [
+                        "nix",
+                        "run",
+                        "-L",
+                        "--impure",
+                        *opts.extra_flags,
+                        "github:NixOS/nixpkgs/nixos-unstable#yq",
+                        "--",
+                        ".",
+                        str(
+                            (
+                                Path(filename).resolve().parent / "pubspec.lock"
+                            ).resolve(),
+                        ),
+                    ],
+                    stdout=f,
+                )
+            (Path(filename).parent / "pubspec.lock").unlink()
+            src = Path(tempdir) / "pubspec.lock.json"
+        else:
+            res = run(
+                [
+                    "nix",
+                    "build",
+                    "--out-link",
+                    f"{tempdir}/result",
+                    "--impure",
+                    "--print-out-paths",
+                    "--expr",
+                    f"\nwith import <nixpkgs> {{}}; \n{get_package(opts)}.overrideAttrs (old: {{\n  nativeBuildInputs = (old.nativeBuildInputs or []) ++ [ pkgs.yq ];\n  postUnpack = ''\n    yq .  \"$sourceRoot/pubspec.lock\" > $out\n    exit\n  '';\n  outputs = [ \"out\" ];\n     separateDebugInfo = false;\n}})\n",
```

Comment:
Make this a multiline string that uses textwrap.dedent: https://docs.python.org/3/library/textwrap.html
This will be more readable.